### PR TITLE
feat : biw save system unity implementation

### PR DESCRIPTION
**Pull request by D4rWiNSS**
# What? 
This implement the scene save system in Unity

# Why? 

Right now you can only save the scene without any feedback when you exit from builder in world. This will implement an auto save feature

### CODE REVIEW

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWFloorHandler.cs [hunk 1]
```diff
@@ -90,15 +90,15 @@ public void CreateFloor(CatalogItem floorSceneObject)
...
             decentralandEntity.rootEntity.OnShapeUpdated += OnFloorLoaded;
-            decentralandEntity.transform.position = WorldStateUtils.ConvertPointInSceneToUnityPosition(initialPosition, parcel);
+            //decentralandEntity.transform.position = ;
             dclBuilderMeshLoadIndicatorController.ShowIndicator(decentralandEntity.rootEntity.gameObject.transform.position, decentralandEntity.rootEntity.entityId);
 
             GameObject floorPlaceHolder = GameObject.Instantiate(floorPrefab, decentralandEntity.rootEntity.gameObject.transform.position, Quaternion.identity);
             floorPlaceHolderDict.Add(decentralandEntity.rootEntity.entityId, floorPlaceHolder);
             decentralandEntity.OnShapeFinishLoading += OnShapeLoadFinish;
-            builderInWorldBridge?.EntityTransformReport(decentralandEntity.rootEntity, sceneToEdit);
+            // builderInWorldBridge?.EntityTransformReport(decentralandEntity.rootEntity, sceneToEdit);
```
Comment (pravusjif) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWFloorHandler.cs]: Can we remove the commented-out code?

**Review by pravusjif — APPROVED** [review]
Also tested the build a bit and saw no problems

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWFloorHandler.cs (hunk 1 above)
Comment (D4rWiNSS) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/Controllers/BIWFloorHandler.cs]: done!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DCLBuilderInWorldEntity.cs [hunk 2]
```diff
@@ -128,11 +130,16 @@ public CatalogItem GetCatalogItemAssociated()
 
     public bool HasShape() { return isShapeComponentSet; }
 
+    public bool HasMovedSinceLastReport() { return Vector3.Distance(lastPositionReported, transform.position) >= BuilderInWorldSettings.ENTITY_POSITION_POSITION_THRESHOLD; }
```
Comment (pbosio) [inline unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DCLBuilderInWorldEntity.cs:133]: Not sure if `HasMovedSinceLastReport` is called frequently, if it does, could we use a squared distance calculation? something like 
```suggestion
    public bool HasMovedSinceLastReport() { return Vector3.SqrMagnitude(lastPositionReported - transform.position) >= BuilderInWorldSettings.ENTITY_POSITION_POSITION_THRESHOLD *  BuilderInWorldSettings.ENTITY_POSITION_POSITION_THRESHOLD; }
```

**Review by pbosio — APPROVED** [review]
LGTM!

File: unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DCLBuilderInWorldEntity.cs (hunk 2 above)
Comment (D4rWiNSS) [reply unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/BuilderEntity/DCLBuilderInWorldEntity.cs:133]: is called each 100 ms and it is only called on the selected entities, so this shouldn't be a problem.

Thanks for the suggestion!

**Pull request merged**